edgar_tool/text_search.py

```diff
@@ -1,48 +1,42 @@
 import itertools
+import re
 import sys
 import urllib.parse
 from datetime import date, timedelta
 from math import ceil
 from typing import List, Optional, Dict, Any, Iterator
 
-from bs4 import BeautifulSoup
-from selenium.webdriver.common.by import By
-from selenium.webdriver.remote.webelement import WebElement
 
-from edgar_tool.browser import (
-    BrowserDriver,
+from edgar_tool.page_fetcher import (
     fetch_page,
-    extract_html_table_rows,
     PageCheckFailedError,
     ResultsTableNotFoundError,
 )
 from edgar_tool.constants import (
     TEXT_SEARCH_BASE_URL,
     TEXT_SEARCH_FILING_CATEGORIES_MAPPING,
-    TEXT_SEARCH_RESULTS_TABLE_XPATH,
     TEXT_SEARCH_SPLIT_BATCHES_NUMBER,
     TEXT_SEARCH_CSV_FIELDS_NAMES,
+    TEXT_SEARCH_FORM_MAPPING,
+    TEXT_SEARCH_LOCATIONS_MAPPING,
 )
 from edgar_tool.io import write_results_to_file
 from edgar_tool.utils import split_date_range_in_n, unpack_singleton_list
 
 
 class EdgarTextSearcher:
 
-    def __init__(self, driver: BrowserDriver):
+    def __init__(self):
         self.search_requests = []
-        self.driver = driver
+        self.json_response = {}
 
     def _parse_number_of_results(self) -> int:
         """
         Parses the number of results found from the search results page.
         :return: Number of results found
         """
-
         num_results = int(
-            self.driver.find_element(By.ID, "show-result-count")
-            .text.replace(",", "")
-            .split(" ")[0]
+            self.json_response.get("hits", {}).get("total", {}).get("value")
```

Comment:
if "value" is None, this will raise an exception, is that accounted for elsewhere in the code?

Reply:
Great spot, the irony is this is the unhandled exception I set out to resolve in the first place!